torchtune/models/clip/_position_embeddings.py

```diff
@@ -0,0 +1,167 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import math
+
+import torch
+from torch import nn
+
+# TODO (Felipe): add load hooks + interpolation on positional encodings,
+# so max_num_tiles can be variable and a trained model can be adapted to a
+# new value.
+
+
+class TokenPositionalEmbedding(nn.Module):
+    """
+    Token positional embedding for images. Different for every token (patch) in an image.
+
+    Args:
+        embed_dim (int): The dimensionality of each token embedding.
+        patch_grid_size (int): The side of a squared grid that represents how many
+             patches are in one tile, i.e. tile_size // patch_size.
+    """
+
+    def __init__(self, embed_dim: int, patch_grid_size: int) -> None:
+        super().__init__()
+        scale = embed_dim**-0.5
+        self.positional_embedding = nn.Parameter(
```

Comment:
IIUC, you don't want to use an ```nn.Embedding``` because you don't actually extract token embeddings from this? We always just add the embeddings to the input tensor because the inputs are of a fixed size?